### ISSUE TITLE
Angular file, undefined is not a function

### ISSUE THREAD

**Issue by madvas** [hormander/json-lzw#3]
There is a problem with angular file. On line 64 `this.JSON.parse(json)` throws error `undefined is not a function` . It should probably be only `JSON.parse(json)`.

After I've tried it I encountered another error with `this` on line 134 `this._indexOf.call` -> `undefined is not a function` , because `this` means something else in `recursiveAstBuilder` function

**Comment by gustavomazzoni** [comment]
Yes, you are right. He created two closures inside `pack` function, `recursiveAstBuilder` and `recursiveParser`. So `this` is not associated to the jsonlzw object anymore, it's associated to window. To use the jsonlzw object inside the closures we need to assign it to a different variable like `self`, for example. Then replace the use of `this` inside the closures to `self`. 

At line 75, add the following code:
```sh
var self = this;
```
Replace `this` to `self` inside the closures like:
```sh
var index = self._indexOf.call(dictionary.strings, item);
```

**Comment by hormander** [comment]
Hi Gustavo
thank you for your interested about my software.
Currently I'm working on other project (non javascript)
If you send me your change, I apply it
For curiosity, where do you use the my library?

PS: excuse me for my bad english

Il 22/07/2016 20:17, Gustavo Mazzoni ha scritto:
> Yes, you are right. He created two closures inside |pack| function, 
> |recursiveAstBuilder| and |recursiveParser|. So |this| is not 
> associated to the jsonlzw object anymore, it's associated to window. 
> To use the jsonlzw object inside the closures we need to assign it to 
> a different variable like |self|, for example. Then replace the use of 
> |this| inside the closures to |self|.

**Comment by gustavomazzoni** [comment]
Hi Marco,

I’ve forked your project and will send you a pull request with the changes I’ve made, ok?

I’m not using your library, actually. I needed some more features and decided to go with lz-string.

**Comment by gustavomazzoni** [comment]
Marco, just created the pull request with the changes. If you have any questions please do it.

**Comment by hormander** [comment]
Hello Gustavo,
Thanks again, I apply now your patch
Tomorrow, I'll update the files on the NPM and, if you want, I'll add 
your name

Best regards,
m

Il 26/07/2016 16:55, Gustavo Mazzoni ha scritto:

**Comment by gustavomazzoni** [comment]
Hi Marco,

You're welcome! Happy to help.
It will be a pleasure to have my name on it.

Best regards,
Gustavo

Em ter, 26 de jul de 2016 às 12:49, Marco notifications@github.com
escreveu:

**Comment by hormander** [comment]
Hi Gustavo,

I added your name on my npm project page 
(https://www.npmjs.com/package/json-lzw)

Best regards,
m

Il 26/07/2016 18:06, Gustavo Mazzoni ha scritto:

**Comment by gustavomazzoni** [comment]
Nice! Thanks!
